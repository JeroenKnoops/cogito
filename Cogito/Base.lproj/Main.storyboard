<?xml version="1.0" encoding="UTF-8"?>
<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="13196" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES" initialViewController="BYZ-38-t0r">
    <device id="retina4_7" orientation="portrait">
        <adaptation id="fullscreen"/>
    </device>
    <dependencies>
        <deployment identifier="iOS"/>
        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="13173"/>
        <capability name="Constraints to layout margins" minToolsVersion="6.0"/>
        <capability name="Safe area layout guides" minToolsVersion="9.0"/>
        <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
    </dependencies>
    <scenes>
        <!--View Controller-->
        <scene sceneID="tne-QT-ifu">
            <objects>
                <viewController id="BYZ-38-t0r" customClass="ViewController" customModule="Cogito" customModuleProvider="target" sceneMemberID="viewController">
                    <view key="view" contentMode="scaleToFill" id="8bC-Xf-vdC">
                        <rect key="frame" x="0.0" y="0.0" width="375" height="667"/>
                        <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
                        <subviews>
                            <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Number of connected peers:" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="cah-t9-UjB">
                                <rect key="frame" x="16" y="20" width="220" height="21"/>
                                <fontDescription key="fontDescription" type="system" pointSize="17"/>
                                <nil key="textColor"/>
                                <nil key="highlightedColor"/>
                            </label>
                            <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="0" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="LnB-ol-chg">
                                <rect key="frame" x="244" y="20" width="11" height="21"/>
                                <fontDescription key="fontDescription" type="system" pointSize="17"/>
                                <nil key="textColor"/>
                                <nil key="highlightedColor"/>
                            </label>
                            <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Sync progress:" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="FpB-lq-rKm">
                                <rect key="frame" x="16" y="49" width="115" height="21"/>
                                <fontDescription key="fontDescription" type="system" pointSize="17"/>
                                <nil key="textColor"/>
                                <nil key="highlightedColor"/>
                            </label>
                            <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="idle" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="Ter-0c-iwG">
                                <rect key="frame" x="139" y="49" width="28" height="21"/>
                                <fontDescription key="fontDescription" type="system" pointSize="17"/>
                                <nil key="textColor"/>
                                <nil key="highlightedColor"/>
                            </label>
                            <progressView opaque="NO" contentMode="scaleToFill" verticalHuggingPriority="750" progress="0.5" translatesAutoresizingMaskIntoConstraints="NO" id="Ck5-ic-qsN">
                                <rect key="frame" x="16" y="78" width="343" height="2"/>
                            </progressView>
                        </subviews>
                        <color key="backgroundColor" red="1" green="1" blue="1" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
                        <constraints>
                            <constraint firstItem="cah-t9-UjB" firstAttribute="top" secondItem="6Tk-OE-BBY" secondAttribute="top" id="2gg-ke-Q0h"/>
                            <constraint firstItem="Ck5-ic-qsN" firstAttribute="leading" secondItem="8bC-Xf-vdC" secondAttribute="leadingMargin" id="BT9-gF-PHv"/>
                            <constraint firstItem="Ck5-ic-qsN" firstAttribute="top" secondItem="FpB-lq-rKm" secondAttribute="bottom" constant="8" id="D3a-xV-vX9"/>
                            <constraint firstItem="FpB-lq-rKm" firstAttribute="top" secondItem="cah-t9-UjB" secondAttribute="bottom" constant="8" id="DAK-LN-QeK"/>
                            <constraint firstItem="FpB-lq-rKm" firstAttribute="leading" secondItem="6Tk-OE-BBY" secondAttribute="leading" constant="16" id="KUl-v7-AaE"/>
                            <constraint firstItem="6Tk-OE-BBY" firstAttribute="trailing" relation="greaterThanOrEqual" secondItem="LnB-ol-chg" secondAttribute="trailing" constant="16" id="OE0-El-ALG"/>
                            <constraint firstItem="6Tk-OE-BBY" firstAttribute="trailing" relation="greaterThanOrEqual" secondItem="Ter-0c-iwG" secondAttribute="trailing" constant="16" id="OHa-nU-llP"/>
                            <constraint firstItem="Ter-0c-iwG" firstAttribute="baseline" secondItem="FpB-lq-rKm" secondAttribute="baseline" id="Uvu-22-vnb"/>
                            <constraint firstItem="LnB-ol-chg" firstAttribute="leading" secondItem="cah-t9-UjB" secondAttribute="trailing" constant="8" id="YWF-fR-kuR"/>
                            <constraint firstItem="Ter-0c-iwG" firstAttribute="leading" secondItem="FpB-lq-rKm" secondAttribute="trailing" constant="8" id="apK-1z-Lie"/>
                            <constraint firstItem="cah-t9-UjB" firstAttribute="leading" secondItem="6Tk-OE-BBY" secondAttribute="leading" constant="16" id="eIS-t7-NAE"/>
                            <constraint firstItem="LnB-ol-chg" firstAttribute="baseline" secondItem="cah-t9-UjB" secondAttribute="baseline" id="jq2-Tb-PbP"/>
                            <constraint firstItem="Ck5-ic-qsN" firstAttribute="trailing" secondItem="8bC-Xf-vdC" secondAttribute="trailingMargin" id="wyS-9z-qYD"/>
                        </constraints>
                        <viewLayoutGuide key="safeArea" id="6Tk-OE-BBY"/>
                    </view>
                    <connections>
                        <outlet property="peerCountLabel" destination="LnB-ol-chg" id="mt8-w3-SWK"/>
                        <outlet property="syncProgressBar" destination="Ck5-ic-qsN" id="TxD-po-cHE"/>
                        <outlet property="syncProgressLabel" destination="Ter-0c-iwG" id="wwa-Hy-aoh"/>
                    </connections>
                </viewController>
                <placeholder placeholderIdentifier="IBFirstResponder" id="dkx-z0-nzr" sceneMemberID="firstResponder"/>
            </objects>
            <point key="canvasLocation" x="136.80000000000001" y="132.68365817091455"/>
        </scene>
    </scenes>
</document>
