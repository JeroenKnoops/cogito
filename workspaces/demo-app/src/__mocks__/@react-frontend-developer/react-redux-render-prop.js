import { WithStoreMock } from '@react-frontend-developer/react-redux-render-prop'

const WithStore = WithStoreMock

export { WithStore }
