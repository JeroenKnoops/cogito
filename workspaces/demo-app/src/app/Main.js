import React from 'react'
import { Switch, Route } from 'react-router-dom'
import { PageCentered } from '@react-frontend-developer/react-layout-helpers'
import { CogitoReact } from '@cogitojs/cogito-react'
import { WithStore } from '@react-frontend-developer/react-redux-render-prop'
import { Dimmer, Loader, Segment } from 'semantic-ui-react'
import { UserDataActions } from 'user-data'

import { Home } from 'pages/home'
import { CogitoId } from 'pages/cogito-id'
import { StreamEncryption } from 'pages/stream-encryption'
import { NoMatch404 } from 'pages/404'

import simpleStorage from 'contracts/SimpleStorage.json'

const contractsInfo = {
  deployedContractsInfo: [
    { contractName: 'simpleStorage', contractDefinition: simpleStorage }
  ],
  rawContractsInfo: []
}

class Main extends React.Component {
  renderComponent = (Component, routeProps, web3Props) => (
    <Component {...web3Props}
      {...routeProps} />
  )

  web3IsReady = ({web3, channel, contracts}) => {
    return (web3 && channel && contracts)
  }

  onTelepathChanged = ({ channelId, channelKey, appName }, dispatch) => {
    dispatch(UserDataActions.setTelepath({ channelId, channelKey, appName }))
  }

  render () {
    return (
      <WithStore
        selector={state => ({
          channelId: state.userData.channelId,
          channelKey: state.userData.channelKey
        })}
        render={({channelId, channelKey}, dispatch) =>
          <CogitoReact contracts={contractsInfo}
            channelId={channelId}
            channelKey={channelKey}
            appName='Cogito Demo App'
            dispatch={dispatch}
            onTelepathChanged={(telepath) => this.onTelepathChanged(telepath, dispatch)}
          >
            {web3Props => {
              if (this.web3IsReady(web3Props)) {
                return (
                  <Switch>
                    <Route exact path='/' render={routeProps => this.renderComponent(Home, routeProps, web3Props)} />
                    <Route exact path='/cogito-id' render={routeProps => this.renderComponent(CogitoId, routeProps, web3Props)} />
                    <Route exact path='/stream-encryption' render={routeProps => this.renderComponent(StreamEncryption, routeProps, web3Props)} />
                    <Route component={NoMatch404} />
                  </Switch>
                )
              } else {
                return (
                  <PageCentered>
                    <div style={{width: '300px', height: '200px'}}>
                      <Segment style={{width: '100%', height: '100%'}}>
                        <Dimmer active inverted>
                          <Loader inverted content={
                            <p>loading web3...</p>
                          } />
                        </Dimmer>
                      </Segment>
                    </div>
                  </PageCentered>
                )
              }
            }}
          </CogitoReact>
        }
      />
    )
  }
}

export { Main }
