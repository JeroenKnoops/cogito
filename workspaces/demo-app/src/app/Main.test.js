import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { Main } from './Main'
import { WithStore } from '@react-frontend-developer/react-redux-render-prop'
import { inRouter } from 'test-helpers/router'

const state = {
  userData: {
    connectionEstablished: false,
    balance: 0
  },
  appEvents: {},
  encryption: {
    plainText: '',
    cipherText: ''
  }
}

WithStore.mockStore(state)

it('renders without crashing', async () => {
  const div = document.createElement('div')
  await ReactDOM.render(inRouter(Main, '/'), div)
})

it('renders correctly', async () => {
  const app = await renderer
    .create(inRouter(Main, '/'))
  expect(app.toJSON()).toMatchSnapshot()
})
