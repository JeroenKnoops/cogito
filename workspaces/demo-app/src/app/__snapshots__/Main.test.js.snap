// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-vm64bh"
>
  <div
    className="css-15ysuyo"
  >
    <div
      className="css-829d7j"
    >
      <div>
        Demo App
      </div>
    </div>
  </div>
  <div
    className="css-1092ap8"
  >
    <div
      className="css-1crpgfv"
    >
      <p>
        Your Cogito account address is:
      </p>
      <div
        className="css-h2wbwd"
      >
        unknown
      </div>
      <p>
        You are known as:
      </p>
      <div
        className="css-h2wbwd"
      >
        unknown
      </div>
    </div>
  </div>
  <div
    className="css-v3skhn"
  >
    <div
      className="css-jbit82"
    >
      <p
        className="css-huraai"
      >
        You have one contract deployed called 
        <span>
          Simple Storage
        </span>
        .
      </p>
      <p
        className="css-huraai"
      >
        Simple storage allows you to store a value in a smart contract.
      </p>
      <p
        className="css-1ej2l17"
      >
        Current value is:
      </p>
      <div
        className="css-h2wbwd"
      >
        0
      </div>
      <p
        className="css-huraai"
      >
        Using the buttons below you can read the currently stored value and also increase it. When increasing the value stored in the contract, you will need to sign the transaction using Cogito identity app. If this is your first transaction, a QR-code will be presented that will let you establish a secure connection between the browser and the iOS Cogito App. You can always request a new QR-code using the Show QR-code button.
      </p>
      <div
        className="css-1gzlhve"
      >
        <button
          className="ui pink basic button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
        >
          Read...
        </button>
        <button
          className="ui pink basic button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
        >
          Increase by 5...
        </button>
        <button
          className="ui pink basic button"
          disabled={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          tabIndex={undefined}
        >
          Show QR code
        </button>
      </div>
    </div>
  </div>
  <div
    className="css-athk3v"
  >
    <div
      className="css-jbit82"
    >
      <p
        className="css-huraai"
      >
        Cogito can encrypt and decrypt data for you. The public-private key pair used for encryption and decryption lives in the user's Cogito mobile app. Since the public key can be known by anyone, it can be retrieved from the Cogito app. Using that public key, encryption is done purely in the browser. Decryption on the other hand requires that the private key is used, so the Cogito mobile app is involved.
      </p>
      <div
        className="css-47eutw"
      >
        <div
          className="css-pepzuv"
        >
          <textarea
            onChange={[Function]}
            onInput={[Function]}
            placeholder="Enter some text"
            rows={3}
            style={
              Object {
                "height": "10em",
                "resize": "",
                "width": "13em",
              }
            }
            value=""
          />
        </div>
        <div
          className="css-1fuv6sj"
        >
          <button
            className="ui pink basic button"
            disabled={undefined}
            onClick={[Function]}
            role="button"
            tabIndex={undefined}
          >
            ―Encrypt→
          </button>
        </div>
        <div
          className="css-gndo50"
        >
          <button
            className="ui pink basic button"
            disabled={undefined}
            onClick={[Function]}
            role="button"
            tabIndex={undefined}
          >
            ←Decrypt―
          </button>
        </div>
        <div
          className="css-b0s7eq"
        >
          <textarea
            onChange={[Function]}
            onInput={[Function]}
            placeholder="Encrypted text appears here"
            rows={3}
            style={
              Object {
                "height": "10em",
                "resize": "",
                "width": "13em",
              }
            }
            value=""
          />
        </div>
      </div>
      <p
        className="css-huraai"
      >
        What actually happens is this: on 
        <em>
          encryption
        </em>
        , a symmetrical key is generated in the browser. This key is used for encrypting the data that is provided. The symmetrical key itself is also encrypted, using the public key from the user's identity. On 
        <em>
          decryption
        </em>
        , the encrypted symmetrical key is sent to the Cogito mobile app, which decrypts it and returns the key itself. That key is then used to decrypt the actual data in the browser. This scheme ensures that the actual data that is encrypted never leaves the browser. Also, the amount of data that is exchanged with the Cogito mobile app is very small (just the symmetrical key).
      </p>
    </div>
  </div>
  <div
    className="css-1j66vpb"
  >
    <div
      className="css-jbit82"
    >
      <p
        className="css-huraai"
      >
        Cogito can store attestations, which are claims about your identity.
      </p>
      <div
        className="css-cv7f2a"
      >
        <div
          className="css-10knj77"
        >
          <div
            className="css-m5rq82"
          >
            <p>
              Scan to add a dummy attestation:
            </p>
            <canvas
              height={128}
              style={
                Object {
                  "height": 128,
                  "width": 128,
                }
              }
              width={128}
            />
          </div>
          <div
            className="css-1crpgfv"
          >
            Your attestation is:
            <div
              className="css-h2wbwd"
            >
              unknown
            </div>
            <button
              className="ui pink basic button"
              disabled={undefined}
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
            >
              Retrieve
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-v8hqy9"
  >
    <div
      className="css-y1c2qe"
    >
      <p
        className="css-15pdwa0"
      >
        © 2018 by Charterhouse
      </p>
    </div>
  </div>
</div>
`;
